### PR TITLE
TRY-JW-98312 Create Group Homepage In Umbraco

### PR DESCRIPTION
**Pull request by JackWCDS**
[*] request sent to umbraco to create a homepage once a group has been created
[*] record inserted into GroupSite table with the groupId and homepage content Id (root website node)
[+] todo: note to add rollback functionality when available

[+] seed data for groupsite table

[+] group homepage content items in umbraco (uSync sync required)

### CODE REVIEW

File: futurenhs.api/FutureNHS.Api/Controllers/ContentController.cs [hunk 1]
```diff
@@ -36,15 +37,15 @@ public async Task<IActionResult> GetImageAsync(Guid imageId)
         [HttpPost]
         [Route("page/{userId:guid}/{groupId}")]
         [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Guid))]
-        public async Task<IActionResult> CreatePageAsync(Guid userId, Guid groupId, [FromBody] CreatePageRequest createRequest, CancellationToken cancellationToken)
+        public async Task<IActionResult> CreatePageAsync(Guid userId, Guid groupId, [FromBody] GeneralWebPageCreateRequest createRequest, CancellationToken cancellationToken)
```
Comment (Matt-Beldon-CDS) [inline futurenhs.api/FutureNHS.Api/Controllers/ContentController.cs:40]: A few unnecessary async state machines in here.

I take it we're going to validate that the user has permission to perform these actions at a later date? currently we're not using `userId` and not validating that that user has permissions to perform these actions

File: futurenhs.api/FutureNHS.Api/Controllers/GroupsController.cs [hunk 2]
```diff
@@ -274,10 +274,10 @@ public async Task<IActionResult> UserLeaveGroupAsync(Guid userId, string slug, C
         }
 
         [HttpGet]
-        [Route("groups/{groupId:guid}/site")]
-        public async Task<IActionResult> GetGroupSiteAsync(Guid groupId, CancellationToken cancellationToken)
+        [Route("groups/{groupSlug:guid}/site")]
+        public async Task<IActionResult> GetGroupSiteAsync(Guid groupSlug, CancellationToken cancellationToken)
```
Comment (Matt-Beldon-CDS) [inline futurenhs.api/FutureNHS.Api/Controllers/GroupsController.cs:278]: if this is meant to use the groups slug, it needs to be of type string

**Review by Tim-Hunt-CDS — APPROVED** [review]

File: futurenhs.api/FutureNHS.Api/DataAccess/Database/Read/GroupDataProvider.cs [hunk 3]
```diff
@@ -435,28 +436,30 @@ public async Task<IEnumerable<GroupMemberDetails>> GetGroupAdminsAsync(string gr
             return member;
         }
 
-        public async Task<GroupSite> GetGroupSiteDataAsync(Guid groupId, CancellationToken cancellationToken)
+        public async Task<GroupSite> GetGroupSiteDataAsync(Guid groupSlug, CancellationToken cancellationToken)
```
Comment (Matt-Beldon-CDS) [inline futurenhs.api/FutureNHS.Api/DataAccess/Database/Read/GroupDataProvider.cs:439]: same here, group slug is a string

File: futurenhs.api/FutureNHS.Api/Controllers/ContentController.cs (hunk 1 above)
Comment (JackWCDS) [reply futurenhs.api/FutureNHS.Api/Controllers/ContentController.cs:40]: Thanks Matt, yes will add these at a later date

**Review by Matt-Beldon-CDS — APPROVED** [review]
Permission checks and async/await overhead will be addressed at a later date

**Pull request merged**